core/src/test/java/org/javarosa/core/model/test/FormDefTest.java

```diff
@@ -432,7 +438,41 @@ public void testSimilarBindConditionsAreDistinguished() throws Exception {
         }
     }
 
-
+    // regression test for when we weren't decrementing multiplicities correctly when repeats were deleted
+    @Test
+    public void testDeleteRepeatMultiplicities() throws IOException {
+        FormParseInit fpi = new FormParseInit("/multiple_repeats.xml");
+        FormEntryController fec = initFormEntry(fpi, "en");
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 1/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 1/2"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 2/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 2/2"));
+        fec.stepToNextEvent();
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 3/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 3/2"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 4/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 4/2"));
+        fec.stepToPreviousEvent();
+        fec.stepToPreviousEvent();
+        fec.deleteRepeat(0);
```

Comment:
What repeat are we deleting here? The first one? or 3/2?
Can you add a check someone to ensure we don't decrement the multiplicity of entries from other repeats? or does it just crash if that code doesn't work properly? If so would still prefer an explicit check, but a comment might also suffice.